app/assets/javascripts/modules/highlight-active-section-heading.js

```diff
@@ -1,148 +1,166 @@
+      this.anchorIDs.push(anchorID)
     }
+  }
 
-    self.getNextHeadingPosition = function ($theNextID) {
-      return $theNextID.offset()
+  HighlightActiveSectionHeading.prototype.updateActiveNavItem = function () {
+    var windowVerticalPosition = this.getWindowPositions()
+    var footer = document.querySelector('.govuk-footer')
+    if (footer) {
+      var footerPosition = this.getFooterPosition(footer)
     }
 
-    self.getFooterPosition = function ($theID) {
-      return $theID.offset()
-    }
+    for (var i = 0; i < this.anchors.length; i++) {
+      var theID = this.anchorIDs[i]
+      var theNextID = this.anchorIDs[i + 1]
 
-    self.getDistanceBetweenHeadings = function (headingPosition, nextHeadingPosition) {
-      var distanceBetweenHeadings = (nextHeadingPosition - headingPosition)
-      return distanceBetweenHeadings
-    }
+      var $theID = document.getElementById(theID.substring(1)) // remove the # at the start
+      if (theNextID) {
+        var $theNextID = document.getElementById(theNextID.substring(1)) // remove the # at the start
+      }
 
-    self.updateActiveNavItem = function () {
-      var windowVerticalPosition = self.getWindowPositions().scrollTop
-      var footerPosition = self.getFooterPosition($('.govuk-footer'))
+      var headingPosition = this.getHeadingPosition($theID)
+      if (!headingPosition) {
+        return
+      }
 
-      $.each(self.$anchors, function (i) {
-        var theID = anchorIDs[i]
-        var theNextID = anchorIDs[i + 1]
+      headingPosition = headingPosition - 53 // fix the offset from top of page
 
-        var $theID = $(theID)
-        var $theNextID = $(theNextID)
+      if (theNextID) {
+        var nextHeadingPosition = this.getNextHeadingPosition($theNextID)
+      }
 
-        var headingPosition = self.getHeadingPosition($theID)
+      var distanceBetweenHeadings = this.getDistanceBetweenHeadings(headingPosition, nextHeadingPosition)
+      var isPastHeading
 
-        if (!headingPosition) {
-          return
-        }
+      if (distanceBetweenHeadings) {
+        isPastHeading = (windowVerticalPosition >= headingPosition && windowVerticalPosition < (headingPosition + distanceBetweenHeadings))
+      } else {
+        // when distanceBetweenHeadings is false (as there isn't a next heading)
+        isPastHeading = (windowVerticalPosition >= headingPosition && windowVerticalPosition < footerPosition)
+      }
 
-        headingPosition = headingPosition.top
-        headingPosition = headingPosition - 53 // fix the offset from top of page
+      if (isPastHeading) {
+        this.setActiveItem(theID)
+      }
+    }
+  }
 
-        if (theNextID) {
-          var nextHeadingPosition = self.getNextHeadingPosition($theNextID).top
-        }
+  HighlightActiveSectionHeading.prototype.getFooterPosition = function (element) {
+    return this.getElementPosition(element)
+  }
```

Comment:
Are we expecting this to ever get called with anything other than `document.querySelector('.govuk-footer')` ?

If not we could probably move all of 

```
    var footer = document.querySelector('.govuk-footer')
    if (footer) {
      var footerPosition = this.getFooterPosition(footer)
    }
```

into here to make the `updateActiveNavItem` method shorter.

Reply:
Unfortunately `getFooterPosition` is needed as it gets overridden in the test. I'll add a comment to explain, as there initially appears to be a lot of redundancy to this.

Reply:
I don't think moving that code into the method would prevent you overriding it?